src/debug/profile.c

```diff
@@ -0,0 +1,181 @@
+            sendPtr = t->histo_base;
+            while (totalBytes > 0) {
+                bytesThisBlock = (totalBytes < 0x800U) ? totalBytes : 0x800;
+
+                ct = 0;
+                while (ct < bytesThisBlock) {
+                    ct += __osRdbSend(sendPtr + ct, bytesThisBlock - ct, RDB_TYPE_GtoH_PROF_DATA);
+                }
+
+                sendPtr += bytesThisBlock;
+                totalBytes -= bytesThisBlock;
+                osRecvMesg(&__osProfAckMQ, NULL, OS_MESG_BLOCK);
+            }
+            t++;
+        }
+   }
+}
+
+void osProfSendWord(u32 word) {
+   u32 ct;
+   u8* sendPtr;
+
+   ct = 0;
+   sendPtr = &word;
+
+    while (ct < sizeof(word)) {
+        ct += __osRdbSend(sendPtr + ct, sizeof(word) - ct, RDB_TYPE_GtoH_PROF_DATA);
+    }
+}
+
+void osProfileFlush(void) {
+    osSendMesg(&__osProfFlushMQ, NULL, OS_MESG_BLOCK);
+}
+
+void osProfileInit(OSProf* profp, u32 profcnt) {
+    u32 i;
+    OSProf* t;
+
+#ifndef NDEBUG
+    if (__osProfileActive) {
+        __osError(ERR_OSPROFILEINIT_STR, 0);
+        return;
+    }
+
+    if (profcnt == 0) {
+        __osError(ERR_OSPROFILEINIT_CNT, 1, profcnt);
+        return;
+    }
+#endif
+
+    for (t = profp; t < profp + profcnt; t++) {
+#ifndef NDEBUG
+        if ((u32)t->histo_base & 1) {
+            __osError(ERR_OSPROFILEINIT_ALN, 1, t->histo_base);
+            return;
+        }
+
+        if (t->text_start >= t->text_end) {
+            __osError(ERR_OSPROFILEINIT_ORD, 2, t->text_start, t->text_end);
+            return;
+        }
+
+        if (((u32)(t->text_end - t->text_start) / 4) > t->histo_size) {
+            __osError(ERR_OSPROFILEINIT_SIZ, 1, t->histo_size);
+            return;
+        }
+#endif
+
+        for(i = 0; i < t->histo_size; i++) {
+            *(t->histo_base + i) = 0;
+        }
+
+    }
+
+    __osProfileActive = FALSE;
+    __osProfileOverflowBin = 0;
+    __osProfileList = profp;
+    __osProfileListEnd = profp + profcnt;
+    __osProfNumSections = profcnt;
+    if (!__osProfileIOActive) {
+        osCreateMesgQueue(&__osProfFlushMQ, &__osProfFlushMesg, 1);
+        osSetEventMesg(OS_EVENT_RDB_FLUSH_PROF, &__osProfFlushMQ, 0);
+        osCreateMesgQueue(&__osProfAckMQ, &__osProfAckMesg, 1);
+        osSetEventMesg(OS_EVENT_RDB_ACK_PROF, &__osProfAckMQ, 0);
+        osCreateThread(&__osProfileIOThread, 0, __osProfileIO, NULL, &__osProfileIOStack[2400], 0x81);
+        osStartThread(&__osProfileIOThread);
+        __osProfileIOActive = TRUE;
+    }
+
+}
+
+void osProfileStart(u32 microseconds) {
+#ifndef NDEBUG
+    if (microseconds < 50) {
+        __osError(ERR_OSPROFILESTART_TIME, 1, microseconds);
+        return;
+    }
+
+    if(__osProfileActive) {
+        __osError(ERR_OSPROFILESTART_FLAG, 0);
+        return;
+    }
+#endif
+
+    osCreateMesgQueue(&__osProfTimerQ, &__osProfTimerMsg, 1);
+    osSetTimer(&__osProfTimer, 0, OS_USEC_TO_CYCLES(microseconds), &__osProfTimerQ, 0);
```

Comment:
```suggestion
    osSetTimer(&__osProfTimer, 0, OS_USEC_TO_CYCLES(microseconds), &__osProfTimerQ, NULL);
```
Last arg is `OSMesg`